### ISSUE TITLE
Linux server use....

### ISSUE THREAD

**Issue by MoJoShMaN** [uhcs2320/hwTwo#4]
I have uploaded my cpp's and .exe as well as several test cases. Code compiles on linux, however ./coogsive "input=_filename_" isnt showing any output. I have the substring var set to 6 in code so it should be checking after "input= for its filename.  Am I not compiling correctly?

**Comment by ShortCircuit865** [comment]
So here's what you do and then let me know if you still have any issues. I'm writing this from step 1 assuming you have nothing setup. Feel free to skip forward to where the steps apply to you.

1) Download a FTP client as it gives you a nice GUI/File Explorer ... I use WinSCP if your on MacOS something similar is bound to exist after a quick google search. https://winscp.net/eng/download.php

2) Download a Zip file of HW1s repository (the green clone/download button right corner) and extract the test cases folder. 
https://github.com/uhcs2320/hw01

3) Login to WinSCP and create a temporary folder named "testing."

4) Put the test cases inside this folder and copy only the CPP file for your program into the test cases folder itself.

5) Open up your choice of a terminal/command prompt/type thing. If you have Putty installed pressing CTRL + P inside of WinSCP launches Putty.

6) Navigate into directory using cd "testing/test-cases"

7) Compile using: g++ -std=c++11 -o coogsive -I ./ *.cpp
//If it successfully compiles now is a good time to type "clear" (no quotation marks) and press enter before the next step

8) Run it against all 20 test cases using: 
for A in a b c d e f g h i j k l m n o p q r s t ; do echo "Test case: $A"; ./coogsive "input=test-${A}.1.txt" | diff - test-${A}.out.txt ; echo "" ; done

9) Have Fun!

Edit: This all only works assuming your code is properly accepting and processing files from the command line. If this works it's no different than manually doing "input=a.1.txt"

If it does NOT working then you have problems with how your accepting parameters in your code.

**Comment by NeoISE** [comment]
@ShortCircuit865 Please don't list out the letters like that, it is just sloppy and could lead to careless mistakes. Use `{a..t}` instead in the condition of the for loop in step 8.

@MoJoShMaN I suspect you are not reading the file correctly; try this to read the file name

``` C++
std::string name(argv[1]);
name.substr(name.find('=') + 1);    // Grab everything after the equal sign
name.substr(0, name.find('\"'));    // remove ending quotes if any
```

**Comment by aleman** [comment]
You have to check your readFile method. 
Cleaning it up a little may help, for example, you define an int variable `i` that is not utilized.
Also, it seems that you are reading the position for `;` but you're not expecting a semicolon as part of the filename. The idea was to **adapt** the example Param.cpp to fit your code.
The example code given to you is not going to solve things for students directly.

I mentioned that I'll looked at your code, but there were things coming up at the office and home, so only now I'm able to give you some suggestions, sorry about that.

I noticed that using global variables is facilitating things in (at least one of) your recursive functions, which is useful. However, if you find getting stuck, then review how global variables are utilized in places such as the readFile method. Global variables are handy but can easily cause trouble due to difficulty of keeping track of where their values can change.

Last comment: I think that the name of the folder should be hw2 instead of hw02.

**Comment by ShortCircuit865** [comment]
@NeoISE I copied the method Meza provided in the first hw0 assignment and simply replace "uhday" with "coogsive" 
So I'm not very experienced with stuff of the sort.

Out of curiosity I tried what I think you were suggesting a few different way without any success.

Are you saying I should wrap the letters in brackets like so? {a, b, c, d, ....}

or literally {a..t}

Ninja Edit: It seems both work and I was making the mistake of wrapping them up in parentheses instead of brackets (facepalm).
